packages/schematics/angular/utility/dependency.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Rule, SchematicContext } from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import * as path from 'path';
+
+const installTasks = new WeakMap<SchematicContext, string[]>();
+
+interface MinimalPackageManifest {
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  peerDependencies?: Record<string, string>;
+}
+
+/**
+ * Adds a package as a dependency to a `package.json`. By default the `package.json` located
+ * at the schematic's root will be used. The `manifestPath` option can be used to explicitly specify
+ * a `package.json` in different location. The type of the dependency can also be specified instead
+ * of the default of the `dependencies` section by using the `type` option with either `dev` or `peer`
+ * values.
+ *
+ * When using this rule, {@link NodePackageInstallTask} does not need to be included directly by
+ * a schematic. A package manager install task will be automatically scheduled as needed.
+ *
+ * @param name The name of the package to add.
+ * @param version The version specifier for the package to add.
+ * @param options An optional object that can contain the `type` of the dependency
+ * and/or a path (`manifestPath`) of a manifest file (`package.json`) to modify.
+ * @returns A Schematics {@link Rule}
+ */
+export function addDependency(
+  name: string,
+  version: string,
+  options: { type?: 'dev' | 'peer'; manifestPath?: string } = {},
+): Rule {
+  const { type, manifestPath = '/package.json' } = options;
+
+  return (tree, context) => {
+    const manifest = tree.readJson(manifestPath) as MinimalPackageManifest;
+
+    const sectionName: keyof MinimalPackageManifest = type ? `${type}Dependencies` : 'dependencies';
+    const dependencySection: Record<string, string> = (manifest[sectionName] ??=
+      Object.create(null));
+
+    if (dependencySection[name] === version) {
+      // Already present
+      return;
+    } else if (dependencySection[name]) {
+      // Update existing version
+      dependencySection[name] = version;
```

Comment:
Question: Could this downgrade the version as well? Do we want to allow that?

Reply:
Another option would be to throw if already present and maybe a flag to allow the behavior.

Reply:
I'll defer to you on the right way to handle this, I don't want to make it more complicated than we need to. Throwing on downgrade or multiple `addDependency()` of the same package might be the most straightforward solution.